k2/csrc/tensor_ops_test.cu

```diff
@@ -325,10 +326,69 @@ static void TestSimpleRaggedIndexSelect1D() {
   }
 }
 
+
 TEST(Index, SimpleRaggedIndexSelect1D) {
   TestSimpleRaggedIndexSelect1D<float>();
   TestSimpleRaggedIndexSelect1D<double>();
   TestSimpleRaggedIndexSelect1D<int32_t>();
 }
 
+
+TEST(Tensor, DiscountedCumSum) {
```

Comment:
Should we make the function a template?
It currently tests only the type `float`. `double` is not tested.

Reply:
i did this. also addressed other comments.  will merge.